src/app/calendar/month-calendar/month-calendar.component.ts

```diff
@@ -0,0 +1,162 @@
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { faAngleLeft, faAngleRight } from '@fortawesome/free-solid-svg-icons';
+import { Store } from '@ngrx/store';
+import { getDay, getDaysInMonth, getWeek } from 'date-fns';
+import {
+  getNextMonth,
+  getPrevMonth,
+  weekDays,
+  monthNames,
+} from 'src/assets/calendarAssets';
+import { IAppointment } from 'src/assets/models/appointment';
+import { IDay } from 'src/assets/models/day';
+import { setDate } from '../calendar-store/calendar.actions';
+
+@Component({
+  selector: 'app-month-calendar',
+  templateUrl: './month-calendar.component.html',
+  styleUrls: ['./month-calendar.component.scss'],
+})
+export class MonthCalendarComponent implements OnInit, OnChanges {
+  arrowLeftIcon = faAngleLeft;
+  arrowRightIcon = faAngleRight;
+  @Input() day = 0;
+  @Input() month = 0;
+  @Input() year = 0;
+  @Input() appointments: IAppointment[] = [];
+  weekDayNames = weekDays;
+  daysInMonth: IDay[] = [];
+  monthNames = monthNames;
+  constructor(private store: Store) {}
+
+  ngOnInit(): void {
+    this.populateDays();
+  }
+
+  ngOnChanges(): void {
+    this.populateDays();
+  }
+  private populateDays(): void {
+    this.daysInMonth = [];
+    const prevMonthDate = getPrevMonth(new Date(this.year, this.month));
```

Comment:
new Date(this.year, this.month) is repeated many times, you can move that to a variable to decrease the number of new Date objects